ui/app/AppLayouts/Chat/panels/communities/JoinPermissionsOverlayPanel.qml

```diff
@@ -0,0 +1,100 @@
+import QtQuick 2.14
+import QtQuick.Controls 2.14
+import QtQuick.Layouts 1.14
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+import StatusQ.Controls 0.1
+
+import AppLayouts.Chat.helpers 1.0
+import AppLayouts.Chat.controls.community 1.0
+
+import utils 1.0
+
+import SortFilterProxyModel 0.2
+
+
+Control {
+    id: root
+
+    property bool joinCommunity: true // Otherwise it means join channel action
+    property bool requirementsMet: true
+    property string communityName
+    property var communityHoldings
+    property string channelName
+    property var viewOnlyHoldings
+    property var viewAndPostHoldings
+    property var moderateHoldings
+
+    signal revealAddressClicked
+
+    QtObject {
+        id: d
+
+        readonly property string communityRequirementsNotMetText: qsTr("Membership requirements not met")
+        readonly property string communityRevealAddressText: qsTr("Reveal your address to join")
+        readonly property string channelRequirementsNotMetText: qsTr("Channel requirements not met")
+        readonly property string channelRevealAddressText: qsTr("Reveal your address to enter")
+
+        function holdingsTextFormat(name, amount) {
+            return CommunityPermissionsHelpers.setHoldingsTextFormat(HoldingTypes.Type.Asset, name, amount)
+        }
+    }
+
+    padding: 35 // default by design
+    spacing: 32 // default by design
+    implicitHeight: column.implicitHeight + topPadding + bottomPadding
+    implicitWidth: column.implicitWidth + leftPadding + rightPadding
+    contentItem: ColumnLayout {
+        id: column
+        spacing: root.spacing
+
+        HoldingsListPanel {
+            Layout.alignment: Qt.AlignHCenter
```

Comment:
To properly adapt for wider content or narrow window, it should be `Layout.fillWidth: true`. Then component should center content on it's own and wrap text if necessary as well.